.github/workflows/pypi-publish.yml

```diff
@@ -0,0 +1,39 @@
+on:
+  push:
+    branches: [ "main" ]
+    paths: ["**.py"]  # only push new packages when python files have changed
```

Comment:
We should publish on any changes in `src/causica` (including config files e.g.), changes to dependencies and any change to a markdown file that is included in the distributed files (e.g. the README)

Reply:
used path-ignore to achieve something similar